src/wazuh_modules/vulnerability_scanner/tests/unit/versionMatcher_test.cpp

```diff
@@ -541,3 +541,89 @@ TEST_F(VersionMatcherTest, compareAmznLinux_casesRealScan)
+}
+
+TEST_F(VersionMatcherTest, matchPEP440)
+{
+    EXPECT_TRUE(VersionMatcher::match("1!2.0b2.post345.dev456", VersionObjectType::PEP440));
+    EXPECT_TRUE(VersionMatcher::match("1.0b2.post345.dev456", VersionObjectType::PEP440));
+
+    EXPECT_FALSE(VersionMatcher::match("1!2.0b2.345.dev456", VersionObjectType::PEP440));
+}
+
+TEST_F(VersionMatcherTest, matchMajorMinor)
+{
+    EXPECT_TRUE(VersionMatcher::match("3.0", VersionObjectType::MajorMinor));
+    EXPECT_TRUE(VersionMatcher::match("1-6", VersionObjectType::MajorMinor));
+
+    EXPECT_FALSE(VersionMatcher::match("3.A", VersionObjectType::MajorMinor));
+    EXPECT_FALSE(VersionMatcher::match("3.0.1", VersionObjectType::MajorMinor));
+    EXPECT_FALSE(VersionMatcher::match("B.3", VersionObjectType::MajorMinor));
+    EXPECT_FALSE(VersionMatcher::match("C-6", VersionObjectType::MajorMinor));
+    EXPECT_FALSE(VersionMatcher::match("7", VersionObjectType::MajorMinor));
+    EXPECT_FALSE(VersionMatcher::match("7.", VersionObjectType::MajorMinor));
+    EXPECT_FALSE(VersionMatcher::match("7-", VersionObjectType::MajorMinor));
+}
+
+TEST_F(VersionMatcherTest, matchSemVer)
+{
+    EXPECT_TRUE(VersionMatcher::match("1.2.2-beta+001", VersionObjectType::SemVer));
+    EXPECT_TRUE(VersionMatcher::match("1.0.1-alpha", VersionObjectType::SemVer));
+    EXPECT_TRUE(VersionMatcher::match("2.1.0-dev", VersionObjectType::SemVer));
+    EXPECT_TRUE(VersionMatcher::match("2.1.0-RC1", VersionObjectType::SemVer));
+    EXPECT_TRUE(VersionMatcher::match("1.2.3", VersionObjectType::SemVer));
+
+    EXPECT_FALSE(VersionMatcher::match("1.2.B", VersionObjectType::SemVer));
+    EXPECT_FALSE(VersionMatcher::match("1:5.15.8-2ubuntu2.0", VersionObjectType::SemVer));
+}
+
+TEST_F(VersionMatcherTest, matchDPKG)
+{
+    EXPECT_TRUE(VersionMatcher::match("1:5.15.8-2ubuntu2.0", VersionObjectType::DPKG));
+    EXPECT_TRUE(VersionMatcher::match("20230206.0~ds2-1.1", VersionObjectType::DPKG));
+    EXPECT_TRUE(VersionMatcher::match("5.15.0-1052.57~20.04.1", VersionObjectType::DPKG));
+
+    EXPECT_FALSE(VersionMatcher::match("21.1-19-gbad84ad4-0ubuntu1~16.04.1", VersionObjectType::DPKG));
+    EXPECT_FALSE(VersionMatcher::match("19.4-56-g06e324ff-0ubuntu1", VersionObjectType::DPKG));
+    EXPECT_FALSE(VersionMatcher::match("1.39+1.40-WIP-2006.11.14+dfsg-2ubuntu1.1", VersionObjectType::DPKG));
+}
+
+TEST_F(VersionMatcherTest, matchRPM)
+{
+    EXPECT_TRUE(VersionMatcher::match("1:4.8.0-2.amzn2023.0.2", VersionObjectType::RPM));
+    EXPECT_TRUE(VersionMatcher::match("2020.3-8.el92", VersionObjectType::RPM));
+    EXPECT_TRUE(VersionMatcher::match("3.1.2-1.el9", VersionObjectType::RPM));
+    EXPECT_TRUE(VersionMatcher::match("20180407-10.el9", VersionObjectType::RPM));
+
+    EXPECT_FALSE(VersionMatcher::match("5.5.6.0_2003-04-09", VersionObjectType::RPM));
+    EXPECT_FALSE(VersionMatcher::match("6.9-11-0", VersionObjectType::RPM));
+}
+
+TEST_F(VersionMatcherTest, matchAMZN)
```

Comment:
Maybe we can add some EXPECT_FALSE cases in this test.

Reply:
Yes. Tried to do so, but is impossible due to #21686.

Reply:
The negative tests were added, after #21686 was closed.